yy_parser.go

```diff
@@ -320,3 +321,35 @@ func getInt64FromNUM(num interface{}) (val int64, errMsg string) {
 	}
 	return -1, fmt.Sprintf("%d is out of range [–9223372036854775808,9223372036854775807]", num)
 }
+
+// convertRoleOrPrivList tries to convert elements of roleOrPrivList to RoleIdentity if toRole, else to PrivElem
+func convertRoleOrPrivList(roleOrPrivList interface{}, toRole bool) (interface{}, error) {
```

Comment:
just split this into two functions.

the input should be `[]*ast.RoleOrPriv` not `interface{}`.